### PR TITLE
[0.10.3] Enrolled Students in Elective Details

### PR DESCRIPTION
**Pull request by SiravitPhokeed**
<div align="center">
  
</div>

## Features
- Enrolled Students in Elective Details Dialog
  Resolves FRO-135 [(link)](https://linear.app/skiso/issue/FRO-135/enrolled-students-in-elective-details-dialog)

## Code Changes
- Extracted Elective Student List component out of Elective Student List Card

## Fixes
- Bottommost Elective List Item obscured by Navigation Bar on mobile
- Cannot select the last selected Elective List Item in Teach Electives on mobile

## Preview
Preview available at https://pr2.mysk.school/.

### CODE REVIEW

File: components/elective/ElectiveStudentList.tsx
Comment (pixelpxed) [inline components/elective/ElectiveStudentList.tsx:1]: Making the list a component makes it great if we need a student list in other pages in the future,

**Review by pixelpxed — APPROVED** [review]
LGTM!

File: components/elective/ElectiveStudentListCard.tsx [hunk 1]
```diff
@@ -114,24 +79,11 @@ const ElectiveStudentListCard: StylableFC<{
...
-                ) : undefined,
-              }}
-              className="w-full !border-0 !bg-surface-container"
-            />
-          ))}
+          <ElectiveStudentList
+            electiveSubject={electiveSubject}
+            query={query}
+            className="[&_button]:!border-0 [&_button]:!bg-surface-container"
+          />
```
Comment (Sun-52) [inline components/elective/ElectiveStudentListCard.tsx:86]: Turning the code into a component is good for making the code more flexible for future use.

**Review by Sun-52 — APPROVED** [review]
LGTM

**Review by smartwhatt — APPROVED** [review]
LGTM!!

**Pull request merged**